src/charts/line.js

```diff
@@ -793,6 +798,56 @@ define(function(require){
                     .attr('y2', height - margin.bottom - margin.top);
         }
 
+        /**
+         * Draws custom user-defined lines onto the chart
+         * @return void
+         */
+        function drawCustomLines(){
+            svg.select('.custom-lines-group')
+                .selectAll('line')
+                .remove();
+
+            let yValues = customLines.map(it => it.y);
+
+            let getColor = yValue => {
+                const definedColor = customLines.find(it => it.y === yValue).color;
+                if(definedColor) {
+                    return definedColor;
+                }
+
+                return 'grey';
+            }
+
+            //draw a horizontal line to extend x-axis till the edges
+            verticalLines = svg.select('.custom-lines-group')
+                .selectAll('line.custom-line')
+                .data(yValues)
+                .enter()
+                .append('line')
+                .attr('class', 'custom-line')
+                .attr('x1', (-xAxisPadding.left - 30))
```

Comment:
Could you clarify why `-xAxisPadding.left - 30`? Wonder how you figured `-30` out.

Reply:
I did not figure it out, it is there for the horizontal grid lines and the base line as well. But yeah, it continues to be a magic number, should have fixed that.

I will create a variable for it.